src/Navigation/Dropdown/Dropdown.test.tsx

```diff
@@ -311,18 +311,15 @@ describe('<Dropdown/> keydown event', () => {
 
   test('handleEscKeydown listener should be removed before component unmount', async () => {
     const removeEventListenerSpy = jest.spyOn(document, 'removeEventListener');
-    const ref = React.createRef<Dropdown>();
     const { unmount } = render(
-      <Dropdown label="Career" ref={ref}>
+      <Dropdown label="Career">
         <DropdownItem value="Product Manager">Product Manager</DropdownItem>
         <DropdownItem value="Software Engineer">Software Engineer</DropdownItem>
       </Dropdown>
     );
 
-    const handleEscKeydown = ref.current.handleEscKeydown;
     unmount();
     expect(removeEventListenerSpy).toHaveBeenCalledTimes(1);
-    expect(removeEventListenerSpy.mock.calls[0][1]).toEqual(handleEscKeydown);
```

Comment:
I wasn't sure how to export the esc event from the functional component to keep this expectation working. Do you think testing that the `removeEventListenerSpy` is sufficient?

Reply:
I think we can do this at least, `listener` is the name of the function returned by `escEvent`:
```
expect((removeEventListenerSpy.mock.calls[0][1] as any).name).toEqual(
  'listener'
);
```

Reply:
Done!